sdk/data/aztables/options.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztables
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	generated "github.com/Azure/azure-sdk-for-go/sdk/data/aztables/internal"
+)
+
+// AddEntityOptions contains optional parameters for Client.AddEntity
+type AddEntityOptions struct {
+	// placeholder for future optional parameters
+}
+
+// CreateTableOptions contains optional parameters for Client.Create and ServiceClient.CreateTable
+type CreateTableOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (c *CreateTableOptions) toGenerated() *generated.TableClientCreateOptions {
+	return &generated.TableClientCreateOptions{}
+}
+
+// DeleteEntityOptions contains optional parameters for Client.DeleteEntity
+type DeleteEntityOptions struct {
+	IfMatch *azcore.ETag
+}
+
+func (d *DeleteEntityOptions) toGenerated() *generated.TableClientDeleteEntityOptions {
+	return &generated.TableClientDeleteEntityOptions{}
+}
+
+// DeleteTableOptions contains optional parameters for Client.Delete and ServiceClient.DeleteTable
+type DeleteTableOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (c *DeleteTableOptions) toGenerated() *generated.TableClientDeleteOptions {
+	return &generated.TableClientDeleteOptions{}
+}
+
+// GetAccessPolicyOptions contains optional parameters for Client.GetAccessPolicy
+type GetAccessPolicyOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (g *GetAccessPolicyOptions) toGenerated() *generated.TableClientGetAccessPolicyOptions {
+	return &generated.TableClientGetAccessPolicyOptions{}
+}
+
+// GetEntityOptions contains optional parameters for Client.GetEntity
+type GetEntityOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (g *GetEntityOptions) toGenerated() (*generated.TableClientQueryEntityWithPartitionAndRowKeyOptions, *generated.QueryOptions) {
+	return &generated.TableClientQueryEntityWithPartitionAndRowKeyOptions{}, &generated.QueryOptions{Format: to.Ptr(generated.ODataMetadataFormatApplicationJSONODataMinimalmetadata)}
+}
+
+// GetPropertiesOptions contains optional parameters for Client.GetProperties
+type GetPropertiesOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (g *GetPropertiesOptions) toGenerated() *generated.ServiceClientGetPropertiesOptions {
+	return &generated.ServiceClientGetPropertiesOptions{}
+}
+
+// GetStatisticsOptions contains optional parameters for ServiceClient.GetStatistics
+type GetStatisticsOptions struct {
+	// placeholder for future optional parameters
+}
+
+func (g *GetStatisticsOptions) toGenerated() *generated.ServiceClientGetStatisticsOptions {
+	return &generated.ServiceClientGetStatisticsOptions{}
+}
+
+// ListEntitiesOptions contains optional parameters for Table.Query
+type ListEntitiesOptions struct {
+	// OData filter expression.
```

Comment:
I'm surprised these docstrings pass the linter

Reply:
I've noticed this before. I suspect that our doc checker isn't properly recursing into struct fields. e.g. some fields in the response envelopes have no doc comments. Opened https://github.com/Azure/azure-sdk-for-go/issues/22446 to track this.